beacon-chain/blockchain/receive_attestation.go

```diff
@@ -83,7 +85,11 @@ func (s *Service) VerifyLmdFfgConsistency(ctx context.Context, a *ethpb.Attestat
 func (s *Service) VerifyFinalizedConsistency(ctx context.Context, root []byte) error {
 	// A canonical root implies the root to has an ancestor that aligns with finalized check point.
 	// In this case, we could exit early to save on additional computation.
-	if s.cfg.ForkChoiceStore.IsCanonical(bytesutil.ToBytes32(root)) {
+	isCanonical, err := s.cfg.ForkChoiceStore.IsCanonical(bytesutil.ToBytes32(root))
+	if err != protoarray.ErrUnknownNodeRoot && err != doublylinkedtree.ErrNilNode {
```

Comment:
If the node is non-canonical or doesnt exist in forkchoice we should continue down. Otherwise old finalized roots will all be rejected here when they are perfectly valid.

Reply:
Same feedback as above